atlassian/github-for-jira
# ARC-1406 navigation between config and select gh version screens

**Rationale:**
We have built most of the UI screens and bits and pieces for the backend so it's time to start wiring it all together. I've started with one point of navigation (from the config page to the GH version selection screen) to keep the number of file changes down. Each subsequent PR will add a new step in the flow.

**Changes:**
See comments: https://github.com/atlassian/github-for-jira/pull/1295/files

**Demo:**
The new path is hidden behind the FF for everyone at this point so no one will wind up on this new flow (expected behaviour).

With FF set to false we still navigate from the config screen to the connect an org screen:

https://user-images.githubusercontent.com/37155488/175880701-f4516df6-96a1-48fd-b061-7e0ccf3660de.mov

With FF set to true (done manually on my local for demo purposes only) we now navigate from the config view to our new GH select version page. We can also navigate back to the config page.

https://user-images.githubusercontent.com/37155488/175880855-be2b051b-3647-44fc-9540-e251fda3b33d.mov

**To test locally:**
In src/routes/jira/configuration/jira-configuration-get.ts change the value from false to true for the FF check for handleNavigationClassName. When the next step of the flow is complete I'll turn the FF on for our segment so we can dogfood before we start rolling out to customers.

**What's up next:**
In ARC 1453 I'll be adding navigation from the GH version select screen 

<img width="909" alt="Screen Shot 2022-06-27 at 8 37 37 pm" src="https://user-images.githubusercontent.com/37155488/175922957-a3d2faa7-0ace-435f-baf4-c7f172548f5c.png">

to the GHE server URL screen (if the user has selected server)

<img width="904" alt="Screen Shot 2022-06-27 at 8 38 07 pm" src="https://user-images.githubusercontent.com/37155488/175923061-afccbbc8-8c7a-460f-9877-d048a2d21ec6.png">

or to continue along the current flow if cloud is selected. Once this is done, I'll have enough to turn on the FF for our segment. We'll still be able to go down the current path, with the addition of the extra screen, and start testing the GHE path as more is added.
